### PR TITLE
Socket Activation for Mariner Package Builds

### PR DESCRIPTION
**Pull request by josephknierman**
Originally Mariner was using the same method to add mgmt and work sockets as centos however this was not necessary since Mariner uses a newer version of systemd that allows for socket activation. This change has the mariner spec file incorporate .socket files and the .service file that requites both socket files.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [X] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [X] Title of the pull request is clear and informative.
- [X] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: builds/mariner/SPECS/aziot-edge/aziot-edge.spec [hunk 1]
```diff
@@ -43,40 +43,31 @@ This package contains the IoT Edge daemon and CLI tool.
...
     unitdir=%{_unitdir} \
     docdir=%{_docdir}/%{name} \
     install
 
+install -D contrib/systemd/mariner/aziot-edged.mgmt.socket %{buildroot}/%{_unitdir}/aziot-edged.mgmt.socket
+install -D contrib/systemd/mariner/aziot-edged.workload.socket %{buildroot}/%{_unitdir}/aziot-edged.workload.socket
+
+# remove initial aziot-edged.service which does not specify mgnt and workload sockets
+rm %{buildroot}/%{_unitdir}/aziot-edged.service
+install -D contrib/systemd/mariner/aziot-edged.service %{buildroot}/%{_unitdir}/aziot-edged.service
```
[the remark this answers is not in the document]
Comment (josephknierman) [reply builds/mariner/SPECS/aziot-edge/aziot-edge.spec]: moved some builds logic to makefile

File: builds/mariner/SPECS/aziot-edge/aziot-edge.spec [hunk 2]
```diff
@@ -43,40 +43,31 @@ This package contains the IoT Edge daemon and CLI tool.
...
+    CONNECT_WORKLOAD_URI=unix:///var/run/iotedge/workload.sock \
+    LISTEN_MANAGEMENT_URI=fd://aziot-edged.mgmt.socket \
+    LISTEN_WORKLOAD_URI=fd://aziot-edged.workload.socket \
     DESTDIR=$RPM_BUILD_ROOT \
     unitdir=%{_unitdir} \
     docdir=%{_docdir}/%{name} \
     install
 
+install -D contrib/systemd/mariner/aziot-edged.mgmt.socket %{buildroot}/%{_unitdir}/aziot-edged.mgmt.socket
+install -D contrib/systemd/mariner/aziot-edged.workload.socket %{buildroot}/%{_unitdir}/aziot-edged.workload.socket
```
[the remark this answers is not in the document]
Comment (josephknierman) [reply builds/mariner/SPECS/aziot-edge/aziot-edge.spec]: moved to make file using $(INSTALL_DATA) -D now

File: builds/mariner/SPECS/aziot-edge/aziot-edge.spec [hunk 3]
```diff
@@ -43,40 +43,31 @@ This package contains the IoT Edge daemon and CLI tool.
 pushd ~
 tar xf %{SOURCE1} --no-same-owner --strip-components=1
 popd
```
[the remark this answers is not in the document]
Comment (josephknierman) [reply builds/mariner/SPECS/aziot-edge/aziot-edge.spec:46]: there was an issue with the implementation here, the version of rpm i tested this section on was newer than the one in ubuntu. this resulted in the version being empty. this should be resolved now.

**Comment by yophilav** [comment]
@arsing Could you help use review this PR please? I want to make sure that we are not missing any details to establish Socket Activation for Mariner OS. Thank you!

**Review by arsing — APPROVED** [review]

**Pull request merged**